### PR TITLE
Add `--factory` and `--seeder` to `twill:make:module`

### PR DESCRIPTION
**Pull request by driftingly**
## Description

This PR adds two new flags to the `twill:make:module` command:
- `--factory`
- `--seeder`

Seeders are made for singletons by default.

### CODE REVIEW

**Comment by ifox** [comment]
Hi @driftingly thanks for your contributions! Excited to release Twill 3.1 with some of them next week!

This is a great addition to the CLI! Could we add capsules compatibility for the factory class to live within the capsule directory? The capsule generator already generates a seeder.

File: src/Commands/ModuleMake.php [hunk 1]
```diff
@@ -633,6 +650,16 @@ private function createModels($modelName = 'Item', $activeTraits = [])
             $activeModelTraitsImports .= "\nuse A17\Twill\Models\Behaviors\Sortable;";
         }
 
+        if ($this->factory && $this->getApplication()->has('make:factory')) {
+            $activeModelTraitsImports .= "\nuse Illuminate\Database\Eloquent\Factories\HasFactory;";
+            $activeModelTraitsString = Str::of($activeModelTraitsString)->replace(';',', HasFactory;');
```
Comment (ifox) [inline src/Commands/ModuleMake.php:655]:
```suggestion
            $activeModelTraitsString = Str::of($activeModelTraitsString)->replace(';', ', HasFactory;');
```

File: src/Commands/ModuleMake.php [hunk 2]
```diff
@@ -633,6 +650,16 @@ private function createModels($modelName = 'Item', $activeTraits = [])
...
+            Artisan::call('make:factory', ['name' => $modelName.'Factory','--model' => $modelClassName], $this->output);
```
Comment (ifox) [inline src/Commands/ModuleMake.php:656]:
```suggestion
            Artisan::call('make:factory', ['name' => $modelName . 'Factory', '--model' => $modelClassName], $this->output);
```

File: src/Commands/ModuleMake.php [hunk 3]
```diff
@@ -633,6 +650,16 @@ private function createModels($modelName = 'Item', $activeTraits = [])
...
+        }
+
+        if ($this->seeder && $this->getApplication()->has('make:seeder')) {
+            Artisan::call('make:seeder '.$modelName.'Seeder', [], $this->output);
```
Comment (ifox) [inline src/Commands/ModuleMake.php:660]:
```suggestion
            Artisan::call('make:seeder ' . $modelName . 'Seeder', [], $this->output);
```

**Review by ifox — CHANGES_REQUESTED** [review]

**Comment by driftingly** [comment]
Hi @ifox Thanks for reviewing so quickly 🚀 
I updated the capsule generator to be compatible with the changes I made to the parent.
I didn't see where it already generates seeders.  If you could show me where I'll update the commands as needed, maybe I'll use the capsule code.

**Comment by ifox** [comment]
Here's where the seeder is generated for capsules: https://github.com/area17/twill/blob/3.x/src%2FCommands%2FModuleMake.php#L315

**Comment by ifox** [comment]
Nice, thanks!

I'm ok with that change, except for the case of singletons. When generating a singleton we do want the seeder to be generated by default because it is automatically called when browsing to that singleton in Twill, to create the single record.

**Comment by driftingly** [comment]
No problem, I’ll update that.

**Comment by driftingly** [comment]
@ifox OK - should be good now.

**Review by ifox — APPROVED** [review]

**Pull request merged**